ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,20 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CXX=${CXX}
+  if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      export CXX="ccache ${CXX}"
+  else
+      export CXX="sccache ${CXX}"
+  fi
```

Comment:
Could you surround this with `if [ -n "${CXX}" ]; then ... fi`? `CXX` may not be defined.